### ISSUE TITLE
Unable to restart GPU

### ISSUE THREAD

**Issue by BarbieAction** [nanopool/ewbf-miner#23]
Running 2x 1080ti cards.

Cards will fail random in couple of days and EWBF miner is unable to restart the GPU's on its own. Need to close and start a new miner for the GPU to load again. This happends around 4 days running all the time.

**Comment by ddurnev** [comment]
The same problem on Maxwell GPU(linux. GTX Titan X): when fails (GPU stuck after too much OC for example) - restart not works, have to manually stop/start the process.

**Comment by blinkin69** [comment]
the same here with 1070 and 1060 GTX

**Comment by dlednik** [comment]
same with GTX1060 cards, they just fall off the bus randonly
sometimes stuck 👎

**Comment by smalinovskiy** [comment]
+1 for 1070. also the --eexit command cannot close miner.

**Comment by dlednik** [comment]
I have temporary solution for my rig. Have 5x 1060 on one motherboard.
I had to set intensity to 0. Now it's been working for couple of days and all cards still mine ok.
I loose about 5% of Sols but at least it works 24/7

**Comment by smalinovskiy** [comment]
It's strange solution, the intensity property have range from 1 to 64. Also setting this property to 0 not help for 1070

**Comment by kennethf541** [comment]
I am experiencing this running 4 PNY 1080's on a Z170A xgaming titanium mobo with a kaby lake 7500hz I5. I believe the cards are crashing out due to my PSU being to small to power them correctly; However, I think the GPU restart is failing as it will continue to mine at 0 sols. A built in check for results higher then 0 sols and an automatic miner restart if it fails would resolve this problem I believe.

**Comment by smalinovskiy** [comment]
This bat helping me to restart miner in 80%. But with delay up to 5 minutes when 0 sol issue happens.
```
:start
timeout /t 60
miner --server eu1-zcash.flypool.org --port 3333 --user *.* --pass 8000 --eexit 3 --intensity 63
echo miner has crashed, restarting...
goto start
```

**Comment by kennethf541** [comment]
I'll give your bat a try. Thanks for sharing it!

**Comment by kennethf541** [comment]
Humm, seems I am still having the issue even using your script. I split them to individual mining sessions to see if that will help. I'll keep you posted.

**Comment by kennethf541** [comment]
Running each card in its own session seems to be working so far. I had the issue happen again but it was once and only crashed 1 card. Here is hoping it continues working until a solution is found.

**Comment by smalinovskiy** [comment]
Yes, running 1-2 cards per bat file is saving resources because restarting of failed GPU takes approximately 5 minutes after the 46 error and the 0 Sols issue. But all other cards works at that time. So I think the root cause of issue that mining not runs really independently for each card.

**Comment by Karlmaster2k3** [comment]
Hello Guys,

hear is my guide for a good mining in ZEC with the EWBF 0.3.4b.

if your gpu stopped, crashed or your internet connection is lost, so take that follow batch.
You got a logfile with the date to directory.
You can make for example 7 gpus 7 batch files with that command -> for the first gpu"--cuda_devices 0" and for second gpu "--cuda_devices 1".... , so if some gpus nevertheless dont restart, your other 6 gpus will run. I will do that today.

Thats my Batch File for  EWBF 0.3.4b CUDA-Miner at my Rigs:

_:restart
TIMEOUT 8
miner --server eu1-zcash.flypool.org --port 3333 --user t1UecXTCCwJ7JqbQDL2SqzmRGQeNYvXT2Zr.rig0 --pass x --eexit 3 --log 2 --logfile ZCASH.%date:~6,4%%date:~3,2%%date:~0,2%.ewbf.log --pec
goto :restart_

- 6 Rigs @ 7 and 8 gpus
- gpu - Asus Strix GTX 1070 08G
- mobo - Asus ROG Strix Z270F Gaming
- cpu - Intel Core i5 6400 2.7 Ghz  - atm i am switching to Intel Celeron G3900 2.8 Ghz
- ssd - SanDisk128GB x400
- ram - Corsair 8GB Vengeance LPX LP DDR4-3200
- psu - 2400 Watt Power Supply Kit for GPU Mining 94%
[https://www.ebay.de/itm/Delta-2400-watt-server-power-supply-Platinum-94-efficeny-Great-for-3-L3s-/332416125933?hash=item4d658b8bed:g:0kUAAOSw3uBZsBXw](url) and some 700-800 Watt power supplies
- Win 10 64bit
- Driver Version 382.53 [http://www.nvidia.de/download/driverResults.aspx/119977/de](url)
- Power Limit 76
- Core Clock 100-120
- Memory Clock 500-600
- Fan Speed 100% it will cost no energie more
![speed2](https://user-images.githubusercontent.com/32879748/31863272-cf26da52-b74b-11e7-9282-7901bf208e33.JPG)
- Energie per gpu arround 123-129 Watt
![watt](https://user-images.githubusercontent.com/32879748/31863221-27b5300c-b74b-11e7-96a2-0fac72fc5118.JPG)
- virtual memory 32GB
![virtuelle-ram](https://user-images.githubusercontent.com/32879748/31863230-4264ae64-b74b-11e7-8862-487c2a8b0a31.JPG)
- Windows creators update [https://support.microsoft.com/de-de/help/4028685/windows-10-get-the-fall-creators-update](url)
- Windows update disabled - Deaktiviert means disabled
![windowsupdate](https://user-images.githubusercontent.com/32879748/31863248-72dd84bc-b74b-11e7-878e-f4b880b7534f.JPG)
- task plan nearly all disabled - Deaktiviert means disabled
![aufgabenplanung](https://user-images.githubusercontent.com/32879748/31863234-47d09c78-b74b-11e7-8fb8-154ff78431f4.JPG)
- automatically restarts rig`s all 8 hours, i will reduce that to one time per day

![gpu_stopped2](https://user-images.githubusercontent.com/32879748/31862768-d435c1fa-b743-11e7-9037-907d0e47094c.JPG)
![gpu_stopped3](https://user-images.githubusercontent.com/32879748/31862769-d6247a4c-b743-11e7-82ba-9ac565029c5a.JPG)
![gpu_stopped4](https://user-images.githubusercontent.com/32879748/31862770-d81f02cc-b743-11e7-948a-0736721dc038.JPG)
![einstellungen](https://user-images.githubusercontent.com/32879748/31862778-ebf5bb56-b743-11e7-8b80-f0565bd5898c.JPG)

Hear you can see my farm ;-)
![img-20171008-wa0004](https://user-images.githubusercontent.com/32879748/31863496-5f3ca43a-b74e-11e7-99b1-084cf683294e.jpeg)
![20171008_175613](https://user-images.githubusercontent.com/32879748/31863497-658101f6-b74e-11e7-8316-4afd70069ea4.jpg)


You Like that Guide? - you can pay some little recognition on follow zcash address : t1UecXTCCwJ7JqbQDL2SqzmRGQeNYvXT2Zr

Best Regards
Karlmaster

**Comment by borzaka** [comment]
In my region (Europe, Hungary) needs a little bit different %DATE% settings:
```
C:\Users\Attila>echo %DATE%
2018. 07. 05.

C:\Users\Attila>echo %DATE:~0,4%%DATE:~6,2%%DATE:~10,2%
20180705
```

So I used:
`--logfile BTG.%DATE:~0,4%%DATE:~6,2%%DATE:~10,2%.ewbf.log`

Thank you for your restart Batch script.
